packages/ember-template-compiler/lib/plugins/index.ts

```diff
@@ -36,8 +34,6 @@ export const STRICT_MODE_TRANSFORMS = Object.freeze(
     TransformActionSyntax,
     TransformEachInIntoEach,
     TransformInElement,
-    AssertSplattributeExpressions,
-    TransformEachTrackArray,
```

Comment:
This is revealing a real issue: https://github.com/emberjs/ember.js/pull/20673

We have to add it back, and deal it somehow on the glimmer-vm side. My initial thought is that glimmer-vm can assume everything that starts with a `-` is a keyword that ember will provide at runtime and not error on those (since it couldn't have been a JavaScript identifier), but maybe that will mask too many real errors when people are migrating existing templates (not uncommon to have helpers starting with dash in apps).

Maybe we'll have to add a `KeywordHead` subnode, or just make it something you are very unlikely to type, like `__GLIMMER_KEYWORD__` or something. Not sure.

But it is a real problem. If we don't deal with it, we would have broken `{{outlet}}`, `{{mount}}`, {{#each-in}}`, `{{#in-element}}` (perhaps among other things) for strict mode templates